core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1072,11 +1071,12 @@ object GroupMetadataManager {
                                         assignment: Map[String, Array[Byte]],
                                         version: Short = 0): Array[Byte] = {
     val value = if (version == 0) new Struct(GROUP_METADATA_VALUE_SCHEMA_V0) else new Struct(CURRENT_GROUP_VALUE_SCHEMA)
+    val protocol = groupMetadata.protocolOrNull
```

Comment:
Can `protocol` be null here? A few lines below `memberMetadata.metadata(protocol)` throws IllegalArgumentException if the protocol is not found.

Reply:
There's an implicit assumption that the protocol is not null if the group is not empty. Let me see if I can find a better way to convey this assumption.